pf/internal/convert/convert_test.go

```diff
@@ -121,6 +120,34 @@ func TestConvertTurnaround(t *testing.T) {
 		), tftypes.List{ElementType: tftypes.Number}, tftypes.String),
 	)
 
+	setCase := func(
+		name string, val tftypes.Type, prop resource.PropertyValue, elements ...any,
+	) convertTurnaroundTestCase {
+		t := tftypes.Set{ElementType: val}
+		return convertTurnaroundTestCase{
+			name: "tftypes.Set/" + name,
+			ty:   t,
+			val:  tftypesNewValue(t, elements),
+			prop: prop,
+		}
+	}
+
+	cases = append(cases,
+		setCase("empty", tftypes.String, resource.NewArrayProperty([]resource.PropertyValue{})),
+		setCase("string-alpha", tftypes.String, resource.NewArrayProperty([]resource.PropertyValue{
+			resource.NewStringProperty("hello"),
+			resource.NewStringProperty("world"),
+		}),
+			"hello",
```

Comment:
I'd be really curious to have these tested:

- what happens if Pulumi passes duplicate items?
- what if some of the items inside are Null, or Unknown, how does that fare? duplicative?
- this seems to indicate order preservation turnaround (interesting), does this property relax to something like order-preserving deduplication?
- TF can't pass duplicate items back to Pulumi right?

Reply:
> I'd be really curious to have these tested:
> 
> * what happens if Pulumi passes duplicate items?

The TFPF framework will error. 

> * what if some of the items inside are Null, or Unknown, how does that fare? duplicative?


> * this seems to indicate order preservation turnaround (interesting), does this property relax to something like order-preserving deduplication?
> * TF can't pass duplicate items back to Pulumi right?

Reply:
Interesting. Well if it's an error that's a bit unfortunate, we might need to backlog a bug with an example of how this error presents to the user. I'm guessing it's going to start with something not user friendly.